src/main/java/com/spikes2212/path/Path.java

```diff
@@ -0,0 +1,25 @@
+package com.spikes2212.path;
+
+import java.util.Arrays;
+import java.util.List;
+
+public class Path {
+    private List<Waypoint> points;
+    public Path(Waypoint... points) {
+        this.points = Arrays.asList(points);
```

Comment:
please put the output of `Arrays.asList(points)` into another list (preferably `LinkedList` due to the nature of the operations we would like to perform on it) because the output of `Arrays.asList` is immutable.